src/itwinai/torch/trainer.py

```diff
@@ -1220,3 +1231,390 @@ def dist_train(
+                # Constant parameters can be added as-is
+                train_loop_config[name] = values
+                continue
+
+            param_type = values.get("type")
+
+            if param_type == "choice":
+                train_loop_config[name] = tune.choice(values["options"])
+
+            elif param_type == "uniform":
+                train_loop_config[name] = tune.uniform(
+                    float(values["min"]), float(values["max"])
+                )
+
+            elif param_type == "quniform":
+                train_loop_config[name] = tune.quniform(
+                    values["min"], values["max"], values["q"]
+                )
+
+            elif param_type == "loguniform":
+                train_loop_config[name] = tune.loguniform(values["min"], values["max"])
+
+            elif param_type == "qloguniform":
+                train_loop_config[name] = tune.qloguniform(
+                    values["min"], values["max"], values["q"]
+                )
+
+            elif param_type == "randint":
+                train_loop_config[name] = tune.randint(values["min"], values["max"])
+
+            elif param_type == "qrandint":
+                train_loop_config[name] = tune.qrandint(
+                    values["min"], values["max"], values["q"]
+                )
+
+            elif param_type == "lograndint":
+                train_loop_config[name] = tune.lograndint(values["min"], values["max"])
+
+            elif param_type == "qlograndint":
+                train_loop_config[name] = tune.qlograndint(
+                    values["min"], values["max"], values["q"]
+                )
+
+            elif param_type == "randn":
+                train_loop_config[name] = tune.randn(values["mean"], values["stddev"])
+
+            elif param_type == "qrandn":
+                train_loop_config[name] = tune.qrandn(
+                    values["mean"], values["stddev"], values["q"]
+                )
+
+            elif param_type == "grid_search":
+                train_loop_config[name] = tune.grid_search(values["options"])
+
+            else:
+                raise ValueError(f"Unsupported search space type: {param_type}")
+
+        return train_loop_config
+
+    # TODO: Can I also log the checkpoint?
+    def checkpoint_and_report(self, epoch, tuning_metrics, checkpointing_data=None):
+        with tempfile.TemporaryDirectory() as temp_checkpoint_dir:
+            checkpoint = None
+
+            should_checkpoint = epoch % self.config.get("checkpoint_freq", 1)
+
+            if checkpointing_data and should_checkpoint:
+                torch.save(checkpointing_data, os.path.join(temp_checkpoint_dir, str(epoch)))
+                checkpoint = train.Checkpoint.from_directory(temp_checkpoint_dir)
+
+        train.report(tuning_metrics, checkpoint=checkpoint)
+
+    def initialize_logger(self, hyperparams: Optional[Dict], rank):
+        print(f"Logger initializing with rank {rank}")
+
+        if self.logger:
```

Comment:
You can negate this. Also, are you sure you want to print the above statement even if `self.logger` is None?

Reply:
Changed this